.pre-commit-config.yaml

```diff
@@ -96,3 +96,17 @@ repos:
           - src
           - --python-version=3.9
         pass_filenames: false
+  - repo: https://github.com/jazzband/pip-tools
+    rev: 7.3.0
+    hooks:
+      - id: pip-compile
+        name: deps
+        alias: deps
+        always_run: true
+        entry: pip-compile -q --no-annotate --output-file=.config/constraints.txt pyproject.toml --all-extras --strip-extras
+        files: ^.config\/.*(requirements|constraints).*$
+        language: python
+        language_version: "3.10" # minimal we support officially
```

Comment:
@ssbarnea creator support Py 3.9 min.

Reply:
@ssbarnea Also, pre-commit CI is failing for the deps hook due to the `language_version` constraint. I don't fully understand how different versions added here would work with pre-commit CI and I see `deps` being skipped in `ansible-dev-tools` (https://github.com/ansible/ansible-dev-tools/blob/main/.pre-commit-config.yaml#L11).

Reply:
I am aware of this, it was too late to finish it yesterday. Working on it.

Regarding py39 support... we dropped it already from most of the other devtools projects. Keep in mind that ansible-core did the same some time ago.

Reply:
Addressing both...